Override New Version of PEEK Calendar

KoraSHughes opened the pull request:
~request for later new semester transition

KoraSHughes committed:
Recloned Template Repo

~readMe imported from main branch

AlphaError commented:
shouldn't be done until we're sure this version of the repo is the one we're going to go with

KoraSHughes committed:
ReadMe .ics Update #1

~ adding info links for .ics files
~ starting to change user story (differentiating baseline vs ideal story)
~ finding more Calendar() and Event() structures in the .ics python API: https://icalendar.readthedocs.io/en/latest/api.html

KoraSHughes committed:
Updated ReadME: User Story & Database Variables

~Cut down on unnecessary vars and structures that aren't integral to the baseline project idea
~Separated user story from advanced features

KoraSHughes committed:
Small ReadMe Clarification

~we only need to import/export finite .ics info, not all of it

KoraSHughes committed:
ReadMe Clarification: User story

~need user accounts to make use of database storing functions

KoraSHughes committed:
ReadMe Update: deleted unnecessary note

KoraSHughes committed:
Updated ReadMe: Edited Function Overview

~ Read/Create/Update/Delete overview

KoraSHughes committed:
Small ReadMe update: class name

KoraSHughes committed:
Small ReadMe Update: Adv. Features

~ added extra data into the "advanced" portion of the story section so we have a better idea of extra variables/funcs to add in case its required for a later assignment

KoraSHughes committed:
Small ReadMe Update: clarification

Bethss committed:
Create test

Bethss committed:
Demo home page

Bethss committed:
Delete test

was testing, not needed anymore, its empty anyways

Bethss committed:
demo landing page

lmk if we should keep this style!

KoraSHughes committed:
Updated ReadME: Create & Read Functions' Design

KoraSHughes committed:
Updated ReadME Formatting

KoraSHughes committed:
Updated ReadMe: finished new function descriptions

KoraSHughes committed:
ReadMe Update: Added UI Overview

KoraSHughes committed:
Small ReadMe Update: UI overview

KoraSHughes committed:
ReadMe Update: Titles Revamp

KoraSHughes committed:
Backed Overhaul 1

~ creating new test dbs jsons
~ creating new data files
--> separating user data code and event data code for simplicity

Bethss committed:
Imported calendar page demo

demo look / idea based on read me!

Bethss committed:
Update README.md

Bethss committed:
Update README.md

Bethss committed:
updated names to AutoCal

Bethss committed:
Updated the group name on UI demos

Bethss committed:
Delete peekcalendar page 1.png

peek name is no longer used

Bethss committed:
Delete peekcalendar.com_events_import__calendar_.png

peek name is no longer used

paul0718 committed:
following tutorial

paul0718 committed:
following tutorial

paul0718 committed:
following tutorial

paul0718 committed:
tutorial done

KoraSHughes committed:
ReadMe Update: Import Page

~specification to import page and related UI

KoraSHughes committed:
Finished User Data

~added all user methods from data.py
~renamed get_users to get_all_users
~created uid generator that just returns a unique random string for now

KoraSHughes committed:
Small User_Data Fixes

~fixing generate_uid

KoraSHughes committed:
More User Data Fixes

~Naming convention stuff

KoraSHughes committed:
Database Datatype Change

~changing the structure of the database so we can access only users or only events or only connections

KoraSHughes committed:
Small User Data Fix 2

~fixing insert into database

KoraSHughes committed:
Starting Event_Data Creation

~ paused to write connection code since is more relevant here

KoraSHughes committed:
Changing User_Data Delete

~actually deletes based on database now

KoraSHughes committed:
Integrated connect_data.py

NOTE: connect db now keys via eid instead of uid!
~into the user and event data I added the connection functions and expounded that to import refrences in the other data files
~I also added get_event and create_event functions

KoraSHughes committed:
Updated Deletion Functions

Bethss committed:
Add files via upload

create event demo

Bethss committed:
Update README.md

added create page demo, made on ps like others (1-2)

Bethss committed:
Create main.yml

GitHub actions test

Bethss committed:
Update main.yml

Bethss committed:
Update data.py

commented demo_home for now ( to use this you must define an environmental variable called demo_home on your pc)

Bethss committed:
Update mongo.sh

Looks like Ethan made one for peek!

Bethss committed:
Update mongo_port.sh

Bethss committed:
Update db_connect.py

swapped callah's with autocal's mongo data

KoraSHughes committed:
ID Spaces to Change in Frontend

KoraSHughes committed:
ReadMe Update: Admit Privs in Design

KoraSHughes committed:
Readme Spacing Updated (small)

KoraSHughes committed:
ReadMe Update: Endpoint Naming

KoraSHughes committed:
Endpoints Update Part 1

KoraSHughes committed:
Small Environ Change

KoraSHughes committed:
Small Event data change

KoraSHughes committed:
Merge branch 'overhauled' of https://github.com/PEEK-NYU/PEEK into overhauled

KoraSHughes committed:
More Endpoint Updates

KoraSHughes committed:
Finished Endpoints Stage 1

KoraSHughes committed:
Prelim Endpoint Tests

KoraSHughes committed:
Flake8 related Cleanup

~cleaned up data files and endpoint code (mostely spacing  and refrence issues)

paul0718 committed:
Set up action secrets

paul0718 committed:
Room Component Code

paul0718 committed:
Fix refresh and naming problems

paul0718 committed:
new buttons

KoraSHughes committed:
Updating Test Endpoints

KoraSHughes committed:
Translating to dict data collection

KoraSHughes committed:
Created User Login Fn

KoraSHughes committed:
Created Session Model

~how an instance of a user might interact with the browser outside of endpoints
~login example

KoraSHughes committed:
Cleaning Up Whitespace Issues

~searched with flake8

KoraSHughes committed:
Adding Context Terminology

~TODO: define state later

KoraSHughes committed:
Small Bugfix

KoraSHughes committed:
Small User Inst Assertion

paul0718 committed:
login form component

paul0718 committed:
login form css, App.js update

Bethss committed:
Update data.py

on mongo we have the following fields for each user:
_id - unique user data id
username - string datatype
password - string datatype
nickname - string datatype
calendar - will contain data structure type that's in sync with .ics icalendar type, probably json (so fields of subfields)

But anyways this is the start of our new mongo database. I will update all database documents from Callahan's examples to ours very soon.

Bethss committed:
Update db_connect.py

Updated mongo database values, will test soon

Bethss committed:
Update event_data.py

Added the actual for reference, we could convert this file to the .ics event manager

Bethss committed:
Update data.py

add_user 
- changed collection to "user_data" since we do not use a "users" collection in our AutoCal db at all
line 111

del_user 
- changed collection to "user_data" since we do not have a "users" collection in our AutoCal db at all
line 111

more updates to come!

Bethss committed:
Update data.py

removed nickname, replaced with email

KoraSHughes committed:
Starting .ics Exploration

KoraSHughes committed:
Merge branch 'overhauled' of https://github.com/PEEK-NYU/PEEK into overhauled

KoraSHughes committed:
More ICS Testing

Bethss committed:
Update event_data.py

Bethss committed:
Update db_connect.py

updating mongo password env variable

paul0718 committed:
Color input & Date input components

paul0718 committed:
App.js update & Text input component

Bethss committed:
Update data.py

Gradually updating db functions, I know we definitely don't need rooming functions & curr unsure abt which files we're deleting as Kora mentioned we'll be doing

KoraSHughes committed:
Testing Extra Calendar Methods

ref https://icalendar.readthedocs.io/en/latest/usage.html

KoraSHughes committed:
Updated iCalendar Tests

Bethss committed:
Update user_data.py

moved things needed, some things are making sense now
I don't think we need the uid generator as currently mongo generates one for each user/entry into our db. I will confirm

KoraSHughes committed:
Updating userdata refrences

KoraSHughes committed:
userdata docstring update

KoraSHughes committed:
Big User Test Update!

KoraSHughes committed:
Linting User/Event Data

KoraSHughes committed:
Small test_data bugfix

Bethss committed:
Update README.md

just emails

KoraSHughes committed:
Taking out Trash (data.py)

~ deleting old refrences

Bethss committed:
Update user_data.py

updating the format here & @ mongo to match the json files and the other files
looks fire now

Bethss committed:
Update users.json

mongo uses '_id' and I can't change it to custom sadly

Bethss committed:
Update events.json

mongo used '_id', sadly can't change to something custom

Bethss committed:
Update connect.json

_id is mongo's unique number for each document in our connection database
and event_id_1 is also there but if we want to use just _id lmk!

Bethss committed:
Update event_data.py

this looks mongo set now, testing later :P

Bethss committed:
Update connect_data.py

so, _id is mongo's unique id, what we'll do is store the _id of event documents in events_id_1 so that they are connected

We will need to edit some methods here, so I'll handle that 👀

KoraSHughes committed:
Starting calendar tests

paul0718 committed:
Prepare to use template

paul0718 committed:
Template Added & Navigation Button Component Replacement

paul0718 committed:
Login/Register Page Basics

paul0718 committed:
Revert "Starting calendar tests"

This reverts commit 8871fcd293b99bf14f782d90cfc73a37f1ba133c.

paul0718 committed:
Revert "Revert "Starting calendar tests""

This reverts commit c99060f36daf7adba7ee859c8ec36d5ce36c8ce0.

paul0718 committed:
Revert "Login/Register Page Basics"

This reverts commit a2a80b5f082f55b069e185bb38aae90860f22b4c.

paul0718 committed:
Revert "Template Added & Navigation Button Component Replacement"

This reverts commit f9e02efd39058d4ff8ea96578a98d1025cc3a91f.

paul0718 committed:
Revert "Prepare to use template"

This reverts commit 46853d5916fb0b6bf7d052eb45d5e94902f81fc0.

paul0718 committed:
Commits reversion succeeds & Prepare for new front end

paul0718 committed:
Submodule problem fixed

paul0718 committed:
Revert "Submodule problem fixed"

This reverts commit bd931e7d07fa7e10de159f9a873ee984e77307b1.

paul0718 committed:
Modifying gitignore

KoraSHughes committed:
*AI Study for the Future

~ study was on "Work-Life Balance" where the output is in a google calendar - if we could get our hands on this, it would greatly improve our endgoal!

Bethss committed:
Update connect.json

_id = mongo's unique id for this connection object
event_id = an existing events id in the db
_user_id = existing user id in the db

Bethss committed:
Update user_data.py

user_id_1 -> _id (mongo uses _id - see line 24)

Bethss committed:
Update connect_data.py

Updating to mongo, I guess we may not need to store unique ids with the name like the # in event_id_1 since each connection object stores one event and then multiple users (or no?)

multiple users would be:
user_id_1, user_id_2, user_id_3, user_id_4, and etc..

Bethss committed:
Update connect.json

 "user_id_1"

Context:
connection will store multiple users; user_id_1, user_id_2, user_id_3 and etc...

Bethss committed:
Update README.md
